keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,541 @@
+Kubernetes Pods. SLAAC's nature is to dynamically assign IP addresses and change
+them over time. Those should be reflected in Pod status. This would not be
+applicable for the Primary PodNetwork.
+
+### Requirements
+
+Below requirements are divided into phases. Each phase will be in its own design
+section of this KEP that will be defined and implemented incrementally.
+
+#### Phase I (base API and reference in Pod)
+1. Multi-network should not change the behavior of existing clusters where no 
+multi-network constructs are configured/used.
+2. Multi-network needs to be represented explicitly as an API construct in
+Kubernetes as it fundamentally ties in with other APIs in the Kubernetes
+networking model, extending their functionality.
+   * We will introduce an PodNetwork API object that represents the existing
+     infrastructure’s networking.
+3. PodNetwork functions as a handle that decouples the implementation
+(i.e. infrastructure/network admins) from the consumer (cluster user). This
+abstraction follows the Kubernetes model of decoupling intent from implementation.
+4. PodNetwork shall not define any implementation specific parameters in its
+specification.
+5. PodNetwork shall provide cluster generic options only.
+6. PodNetwork can reference to implementation-specific parameters.
+7. PodNetwork is the object Application developer will reference in their
+workflows (e.g. Pod, Service).
+8. Cluster Default PodNetwork is the PodNetwork the cluster has been created
+with initially.
+9. Cluster Default PodNetwork cannot be removed, but can be replaced.
+10. Pods can reference one or more PodNetworks when trying to attach to them.
+    * This list is explicit and all the PodNetworks that has to be available in
+      the Pod has to be listed.
+11. When Pods does not specify any reference to PodNetwork it connects to Cluster
+Default PodNetwork (network the cluster has been created with).
+12. Pod shall be able to provide additional configuration on how it attaches to
+a PodNetwork.
+13. Every Pod connected to specific PodNetwork must have connectivity within
+that network across the Cluster.
+14. A Pod connected to specific PodNetwork may or may not have cross 
+connectivity between different PodNetworks.
+15. Pods attached to PodNetworks are connected to each other in a manner 
+defined by the PodNetwork implementation.
+
+#### Phase II (scheduler, kubelet and API probing for PodNetwork)
+16. Pods access to PodNetwork is controlled via RBAC configuration.
+17. Implementation-agnostic PodNetwork Interface information (e.g. PodNetowrk
+name, IP address, etc.) for each attachment will be exposed to runtime Pod (via 
+e.g. environment variables, downward API etc.).
+18. PodNetworks can be selectively (per Node) available in the Cluster. This
+does NOT apply to the Cluster Default PodNetwork.
+19. Kubelet network-based probing can be optionally enabled for the non- Cluster
```

Comment:
Network probe is defined by workload, why do we need to control enable/disable corresponding to the PodNetwork attachments?

Reply:
We need to ensure it could work with the pod-specified configuration and if we need to tweak kubelet in anyway to achieve this easily.